pkg/trace/graphbuilder_test.go

```diff
@@ -0,0 +1,365 @@
+					"metadata": map[string]interface{}{
+						"name": "standard",
+					},
+					"reclaimPolicy": "Delete",
+				}},
+				n: &Node{
+					Instance: &unstructured.Unstructured{Object: map[string]interface{}{
+						"kind":       "StorageClass",
+						"apiVersion": "storage.k8s.io/v1",
+						"metadata": map[string]interface{}{
+							"name": "standard",
+						},
+					}},
+					GVR: schema.GroupVersionResource{
+						Group:    "storage.k8s.io",
+						Version:  "v1",
+						Resource: "storageclass",
+					},
+					Relateds: nil,
+					State:    "",
+				},
+			},
+			want: want{
+				err: nil,
+			},
+		},
+	}
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			fc := tc.client
+			g := &KubeGraphBuilder{
+				client: fc,
+			}
+			i := tc.instanceToCreate
+			var createdInstance *unstructured.Unstructured
+			if i != nil {
+				var err error
+				if i.GetNamespace() == "" {
+					createdInstance, err = fc.Resource(tc.n.GVR).Create(i, metav1.CreateOptions{})
+				} else {
+					createdInstance, err = fc.Resource(tc.n.GVR).Namespace(i.GetNamespace()).Create(i, metav1.CreateOptions{})
+				}
+
+				if err != nil {
+					t.Fatalf("failed to prepare fake env, could not create Instance: %v", err)
+				}
+			}
+			gotErr := g.fetchObj(tc.args.n)
+
+			if diff := cmp.Diff(tc.want.err, gotErr, test.EquateErrors()); diff != "" {
+				t.Fatalf("g.fetchObj(...): -want error, +got error: %s", diff)
+			}
+			if tc.err != nil {
+				return
+			}
+			gotResult := tc.args.n.Instance
+			if diff := cmp.Diff(createdInstance, gotResult); diff != "" {
+				t.Errorf("g.fetchObj(...): -want result, +got result: %s", diff)
+			}
+		})
+	}
+}
+
+func TestKubeGraphBuilder_BuildGraph(t *testing.T) {
+	scheme := runtime.NewScheme()
+
+	computeGV := schema.GroupVersion{
+		Group:   "compute.crossplane.io",
+		Version: "v1alpha1",
+	}
+
+	gcpGV := schema.GroupVersion{
+		Group:   "compute.gcp.crossplane.io",
+		Version: "v1alpha2",
+	}
+
+	rm := meta.NewDefaultRESTMapper([]schema.GroupVersion{
+		computeGV,
+	})
+	rm.Add(computeGV.WithKind("kubernetescluster"), meta.RESTScopeNamespace)
+	rm.Add(gcpGV.WithKind("gkecluster"), meta.RESTScopeNamespace)
+
+	type args struct {
+		client            dynamic.Interface
+		restMapper        meta.RESTMapper
+		gvrs              []schema.GroupVersionResource
+		instancesToCreate []*unstructured.Unstructured
+		name              string
+		namespace         string
+		resource          string
+	}
+	type want struct {
+		traversed []*Node
+		err       error
+	}
+	cases := map[string]struct {
+		args
+		want
+	}{
+		"NotExist": {
+			args: args{
+				client:            fake.NewSimpleDynamicClient(scheme),
+				restMapper:        rm,
+				instancesToCreate: nil,
+				name:              "test",
+				namespace:         "testnamespace",
+				resource:          "kubernetescluster",
+			},
+			want: want{
+				err: errors.NewNotFound(schema.ParseGroupResource("kubernetesclusters.compute.crossplane.io"), "test"),
```

Comment:
When I ran the unit tests, I got this:

```
--- FAIL: TestKubeGraphBuilder_BuildGraph (0.00s)
    --- FAIL: TestKubeGraphBuilder_BuildGraph/NotExist (0.00s)
        graphbuilder_test.go:306: g.BuildGraph(...): -want error, +got error:   interface{}(
            - 	e`kubernetesclusters.compute.crossplane.io "test" not found`,
            + 	e`Object to trace is not found: "kubernetescluster" "test" in namespace "testnamespace"`,
              )
```

Reply:
Fixed. I should get used to running make test before pushing :)

Reply:
haha : ) - ideally we'll also set up some automated builds so we don't need to always remember